src/values/ObjectValue.js

```diff
@@ -290,6 +299,19 @@ export default class ObjectValue extends ConcreteValue {
     return this._isPartial.value;
   }
 
+  makeTainted(): void {
+    this._isTainted = this.$Realm.intrinsics.true;
+    // Clear all properties. This object is now completely unknown.
+    this.properties.clear(); // TODO: Track that this happens so that it can be undone by evaluateForEffects
+    // TODO: Clear symbol properties, prototype and other fields as well.
+    this.makePartial();
+    this.makeSimple(); // TODO: Don't assume it is once property access no longer fatal.
```

Comment:
"once property access no longer fatal". Do you have an idea how to handle this? I don't have a clear vision for this problem.

Reply:
The idea I had was that invoking an unknown getter/setter is just the same as invoking an unknown function. So if you do that on a leaked object, it would generate an effect just like calling a function on it would. The object would "leak" again so any properties that are known at that time would be reset again.

For setters, it would also leak the value passed into the setter.